Adding text style navigation property

Alfonso-Martello opened the pull request:
### Changes
-	This PR removes the `styleId` property from `TextBlock` and adds a new navigation property called `defaultTextStyle` to the `TextAnnotation2d/3d` elements. This property is optional, since I think connectors are currently writing TextAnnotation elements and requiring the property would be breaking. Uses bis changes from https://github.com/iTwin/bis-schemas/pull/563.
-	This PR simplifies the `TextStyleResolver` by separating `scaleFactor` logic. This is to allow workflows where we can create `TextAnnotation`s before we create a drawing. An example of such a workflow is using AI to generate text but only writing the drawing and the text elements to the iModel once the AI process completes. We still expect `AnnotationTextStyles` to be inserted before any text creation happens.
-	This PR marks the `TextAnnotation2d/3d.create`methods as `beta` and introduces new “create” interfaces so these methods can accept an object as an argument, making new arguments easier to add in the future. I am hoping this is okay, technically this is a method going from public to beta, but it was introduced recently (probably should have been beta from the start) and is part of text, which theoretically is only being used by Drawing Production at this point.

### Issues
-	extract-api is unhappy because the new `ITextAnnotationProps` interface is `beta` and `TextAnnotation2d/3dProps` is `public`. Looking for advice on the best way to proceed (aka "Make ITextAnnotationProps public", "Forget about ITextAnnotationProps and simply add defaultTextStyle as a beta property of each Props that needs it", etc)

diegoalexdiaz commented:
ECDb doesn't enforce _mandatory_ navigation-properties today, even if its target multiplicity is set to 1..1. Such multiplicity will be useful for a later data-validation exercise, but not enforce at data-writing time.

Having said that, I agree that the target multiplicity of the nav-property should be modified to be 0...1.

pmconne commented:
`TextAnnotationUsesTextStyleByDefault` is a relationship between an `ITextAnnotation` and an `AnnotationTextStyle`.
Shouldn't you add `defaultTextStyleId` to the TypeScript `ITextAnnotation` interface?

Alfonso-Martello commented on core/backend/src/annotations/TextBlockLayout.ts:
```diff
@@ -201,37 +200,26 @@ function applyBlockSettings(target: TextStyleSettings, source: TextStyleSettings
   iModel: IModelDb;
-  /** The ID of the model containing the text block, used to compute the scale factor. */
-  modelId?: Id64String;
   /** @internal chiefly for tests, by default looks up an [[AnnotationTextStyle]] in the iModel by ID. */
```
It is marked as `internal` because it really is just used for testing, there is no production use case as of right now

pmconne commented on core/common/src/ElementProps.ts:
```diff
@@ -250,11 +250,20 @@ export interface GeometricElement2dProps extends GeometricElementProps {
   typeDefinition?: RelatedElementProps;
 }
 
+/**
+ * Properties of an ITextAnnotation.
+ * @beta
+ */
+export interface ITextAnnotationProps {
+  /** The default [[AnnotationTextStyle]] used by the annotation. */
```
```suggestion
/**
 * JSON representation of an [[ITextAnnotation]].
 * @beta
 */
export interface ITextAnnotationProps {
  /** The default [AnnotationTextStyle]($backend) element used by the annotation. */
```

pmconne commented on core/common/src/annotation/TextBlock.ts:
```diff
@@ -732,13 +723,8 @@ export class TextBlock extends TextBlockComponent {
-  /** Create an empty text block containing no [[paragraphs]] and an empty [[styleId]]. */
-  public static createEmpty(): TextBlock {
-    return TextBlock.create({ styleId: "" });
+  public static create(props?: TextBlockProps): TextBlock {
+    return new TextBlock(props ?? {});
```
This is a silly suggestion, please ignore it.

Alfonso-Martello commented:
> Just don't include the implements clause, you don't need it in TypeScript.

Removed the `ITextAnnotationProps` interface and just added the `defaultTextStyle` property as `beta` to the interfaces that use it
